### PR TITLE
AND-36

### PR DESCRIPTION
**Pull request by lpetka**

### CODE REVIEW

File: presentation/src/main/java/com/activity_sync/presentation/presenters/AllEventsPresenter.java [hunk 1]
```diff
@@ -25,16 +25,18 @@ public AllEventsPresenter(IEventsFragmentView view, INavigator navigator, Schedu
     @Override
     public void start()
     {
-        if (view.checkLocationPermissions() == false)
+        if (isCurrentlyShown())
```
Comment (mzielu) [inline presentation/src/main/java/com/activity_sync/presentation/presenters/AllEventsPresenter.java]: what if view is not currently shown and has permissions?

File: app/src/main/java/com/activity_sync/screens/MyEventsFragment.java [hunk 2]
```diff
@@ -13,4 +14,10 @@
     protected IPresenter createPresenter(FragmentScreen screen, Bundle savedInstanceState) {
         return new MyEventsPresenter(this, navigator, AndroidSchedulers.mainThread(), apiService);
     }
+
+    @Override
+    public int getIndex()
+    {
+        return 1;
```
Comment (mzielu) [inline app/src/main/java/com/activity_sync/screens/MyEventsFragment.java]: create in EventsFragmentBase protected static int MY_EVENTS_INDEX = 1;

File: app/src/main/java/com/activity_sync/screens/AllEventsFragment.java [hunk 3]
```diff
@@ -15,4 +15,10 @@ protected IPresenter createPresenter(FragmentScreen screen, Bundle savedInstance
     {
         return new AllEventsPresenter(this, navigator, AndroidSchedulers.mainThread(), apiService);
     }
+
+    @Override
+    public int getIndex()
+    {
+        return 0;
```
Comment (mzielu) [inline app/src/main/java/com/activity_sync/screens/AllEventsFragment.java]: create in EventsFragmentBase protected static int ALL_EVENTS_INDEX = 0;

File: app/src/main/java/com/activity_sync/screens/AllEventsFragment.java [hunk 4]
```diff
@@ -15,4 +15,10 @@ protected IPresenter createPresenter(FragmentScreen screen, Bundle savedInstance
     {
         return new AllEventsPresenter(this, navigator, AndroidSchedulers.mainThread(), apiService);
     }
+
+    @Override
+    public int getIndex()
```
Comment (mzielu) [inline app/src/main/java/com/activity_sync/screens/AllEventsFragment.java]: change method name to "getCurrentFragmentIndex"

File: presentation/src/main/java/com/activity_sync/presentation/presenters/AllEventsPresenter.java (hunk 1 above)
Comment (lpetka) [reply presentation/src/main/java/com/activity_sync/presentation/presenters/AllEventsPresenter.java]: then prompt is not shown

Comment (mzielu) [reply presentation/src/main/java/com/activity_sync/presentation/presenters/AllEventsPresenter.java]: but super.start() is not firing up, this is not a good sign

Comment (lpetka) [reply presentation/src/main/java/com/activity_sync/presentation/presenters/AllEventsPresenter.java]: Ok, I see now

**Pull request merged**